src/main/java/de/tarent/resources/CoursesResource.java

```diff
@@ -78,7 +78,7 @@ public void delete(@PathParam("id") Long id) {
                 });
     }
 
-    private Predicate<Course> notDeleted() {
+    static Predicate<Course> notDeleted() {
```

Comment:
see above

Reply:
what is the reason behind this weird predicate method?
Should be `@JsonIgnore public boolean isNotDeleted() {...}` and then the predicate is Course::isNotDeleted

Reply:
how should that work with Optional.filter(Predicate)

Reply:
with `@JsonIgnore public boolean isNotDeleted() {...}` that would be `myOptional.filter(Course::isNotDeleted)`